File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map < int,int>  mp;
        // mp   key = val.nums | val = index.nums
        for (int i = 0; i < nums.size(); i++){
            int complement = target - nums[i];
            if (mp.find(complement) != mp.end())
                return {i, mp[complement]};
            
            mp[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int s = nums.Length;
        for (int i = 0; i < s; i++) {
            for (int j = i + 1; j < s; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[] {i, j};
                }
            }
        }
        return new int[0]; // or throw an exception
    }
}